flutter/lib/ui/geometry.dart

```diff
@@ -680,35 +671,102 @@ class Rect {
   String toString() => "Rect.fromLTRB(${left.toStringAsFixed(1)}, ${top.toStringAsFixed(1)}, ${right.toStringAsFixed(1)}, ${bottom.toStringAsFixed(1)})";
 }
 
-// A rounded rectangle with the same radii for all four corners.
+/// A radius for either circular or oval shapes.
+class Radius {
+  /// Constructs a circular radius.
+  const Radius.circular(double radius) : this.oval(radius, radius);
+
+  /// Construct an oval radius.
+  const Radius.oval(double this.x, double this.y);
+
+  /// The radius value on the horizontal x axis.
+  final double x;
+
+  /// The radius value on the vertical y axis.
+  final double y;
+}
```

Comment:
please add an operator==, hashCode, and toString to this class.

For operator== and hashCode, copy the implementations from elsewhere in this file.